conformance/utils/kubernetes/helpers.go

```diff
@@ -636,12 +635,20 @@ func listenersMatch(t *testing.T, expected, actual []v1beta1.ListenerStatus) boo
 		// Ensure that the expected Listener.SupportedKinds items are present in actual Listener.SupportedKinds
 		// Find the items instead of performing an exact match of the slice because the implementation
 		// might support more Kinds than defined in the test
-		eSupportedKindsSet := sets.New(eListener.SupportedKinds...)
-		aSupportedKindsSet := sets.New(aListener.SupportedKinds...)
-		if !aSupportedKindsSet.IsSuperset(eSupportedKindsSet) {
-			t.Logf("Expected %v kinds to be present in SupportedKinds", eSupportedKindsSet.Difference(aSupportedKindsSet))
-			return false
+		for _, eKind := range eListener.SupportedKinds {
+			found := false
+			for _, aKind := range aListener.SupportedKinds {
+				if *eKind.Group == *aKind.Group && eKind.Kind == aKind.Kind {
+					found = true
+					break
+				}
+			}
+			if !found {
+				t.Logf("Expected Group:%s Kind:%s to be present in SupportedKinds", *eKind.Group, eKind.Kind)
```

Comment:
It's a bit of a nit, but in multiple places here we're de-referencing `Group` and the attribute is technically optional and _could_ be `nil`.

We colloquially expect any `RouteGroupKinds` that come through here will have been initialized via the API first so generally we wouldn't expect this to be an issue because of CRD defaults that would be applied. That said, some situation where someone later tries to use these helper functions in a more derived way would leave them with a segmentation fault that might be little terse.

One thing you could do would be just initialize them with the default value before using them if needed:

```diff
diff --git a/conformance/utils/kubernetes/helpers.go b/conformance/utils/kubernetes/helpers.go
index a0926f9d..33a0c7a8 100644
--- a/conformance/utils/kubernetes/helpers.go
+++ b/conformance/utils/kubernetes/helpers.go
@@ -638,6 +638,14 @@ func listenersMatch(t *testing.T, expected, actual []v1beta1.ListenerStatus) boo
                for _, eKind := range eListener.SupportedKinds {
                        found := false
                        for _, aKind := range aListener.SupportedKinds {
+                               if eKind.Group == nil {
+                                       eKind.Group = (*v1beta1.Group)(&v1beta1.GroupVersion.Group)
+                               }
+
+                               if aKind.Group == nil {
+                                       aKind.Group = (*v1beta1.Group)(&v1beta1.GroupVersion.Group)
+                               }
+
                                if *eKind.Group == *aKind.Group && eKind.Kind == aKind.Kind {
                                        found = true
                                        break
```

Reply:
since `Group` is optional, and if you believe GAPI code might skip populating `Group`, then it makes more sense for the `Group` check to be skipped imho

Reply:
I don't think _skipping_ the group check would be right, because if you did that you could have two kinds of the same name, with two different group names that would be considered matching right?

Reply:
also previously the check was an exact Match and all implementations abided by it, will make this change if thats your preference

Reply:
Unless I've missed something, this just means we'll provide the already configured default behavior in edge cases rather than a segfault, so yeah if it's not a problem I would appreciate it thank you.

Reply:
sg, will add